code/controllers/subsystem/tts.dm

```diff
@@ -0,0 +1,239 @@
+/*!
+ * Copyright (c) 2020 Aleksej Komarov
+ * SPDX-License-Identifier: MIT
+ */
+
+/proc/tts_alphanumeric_filter(text)
+	// Only allow alphanumeric characters and whitespace
+	var/static/regex/bad_chars_regex = regex("\[^a-zA-Z0-9 ,?.!'&-]", "g")
+	return bad_chars_regex.Replace(text, " ")
+
+
+#define TARGET_INDEX 1
+#define IDENTIFIER_INDEX 2
+#define TIMEOUT_INDEX 3
+#define REQUEST_INDEX 4
+#define MESSAGE_INDEX 5
+#define EXTRA_TARGETS_INDEX 6
+#define LANGUAGE_INDEX 7
+
+#define CACHE_PENDING "pending"
+
+SUBSYSTEM_DEF(tts)
+	name = "Text To Speech"
+	// Basically every tick
+	wait = 0.05 SECONDS
+	init_order = INIT_ORDER_TTS
+	priority = FIRE_PRIORITY_TTS
+
+	/// Queued HTTP requests that have yet to be sent
+	var/list/queued_tts_messages = list()
+
+	/// Queued HTTP requests that have yet to be sent. Takes priority over queued_tts_messages
+	var/list/priority_queued_tts_messages = list()
+
+	/// HTTP requests currently in progress but not being processed yet
+	var/list/in_process_tts_messages = list()
+
+	/// HTTP requests that are being processed to see if they've been finished
+	var/list/current_processing_tts_messages = list()
+
+	/// A list of available speakers
+	var/list/available_speakers = list()
+
+	var/list/cached_voices = list()
+
+	/// Whether TTS is enabled or not
+	var/tts_enabled = FALSE
+
+	var/message_timeout = 5 SECONDS
+
+	/// Messages can be timed out earlier if the algorithm thinks that
+	/// it's going to take too long for their message to be processed.
+	/// This'll determine the minimum extent of how late it is allowed to begin timing messages out
+	var/message_timeout_early_minimum = 3 SECONDS
+
+	var/max_concurrent_requests = 15
```

Comment:
```suggestion
	var/max_concurrent_requests = 5
```

at 2 threads per worker this is 2 requests processing, and 1.5 in the queue for each thread.